src/server/db/actions/GameAction.ts

```diff
@@ -72,70 +72,124 @@ export async function editGame(allData: nextEditGame) {
-      tagResults.forEach((result, index) => {
-        if (!result) {
-          throw new GenericUserErrorException(
-            `ObjectID ${data.tags![index]} is not a  present tag.`,
-          ); //Using non-null assertion, as if condition should ensure data.tags is non-null
-        }
+  }
+  if (data && data.tags) {
+    const tagResults = await TagModel.find({ id: { $in: data.tags } });
+    if (tagResults.length !== data.tags.length) {
+      throw new GenericGameErrorException(
+        "One of the given tags does not exist.",
+      ); //Using non-null assertion, as if condition should ensure data.tags is non-null
+    }
+  }
+
+  const result = await GameModel.findByIdAndUpdate(allData.id, allData.data, {
+    new: false,
+  });
+  if (!result) {
+    throw new GenericGameErrorException("Game with given ID does not exist.");
+  }
+  return result;
+}
+export async function getSelectedGames(
+  query: z.infer<typeof GetGameQuerySchema>,
+) {
+  await connectMongoDB();
+  //Want to be able to filter by specific tags,game_builds, etc.
+  //Find the corresponding id to the Theme tag.
+  let filters: FilterQuery<IGame> = {}; //There should be a better way to build the query gradually
+  if (query.theme) {
+    //Find corresponding themeObject in mongodb
+    const found_theme: (ITheme & { id: ObjectId }) | null =
+      await ThemeModel.findOne({
+        name: query.theme,
       });
+    if (!found_theme) {
+      //No theme was found
+      throw new GenericGameErrorException(
+        `No theme with the name ${query.theme} exists.`,
+      );
     }
-  } catch (e) {
-    throw e;
+    //Add to filters object
+    filters.themes = { $all: [found_theme] };
+    //Essentially ensures that the themes array contains found_theme.
   }
-  try {
-    const result = await GameModel.findByIdAndUpdate(allData.id, allData.data, {
-      new: false,
+  //Find the correspoding tag ids for the accessibility TAGS.
+  let accessibility: ObjectId[] = [];
+  if (query.accessibility) {
+    const accessibilityTags = await TagModel.find({
+      name: { $in: query.accessibility },
+      type: "accessibility",
     });
-    if (!result) {
-      throw new GenericUserErrorException("Game with given ID does not exist.");
+    //Ensure that we found all tags and no extras
+    if (query.accessibility.length !== accessibilityTags.length) {
+      throw new GenericGameErrorException(
+        "One or more of the accessibility tags do not exist.",
+      );
     }
-    return result;
-  } catch (e) {
-    throw e;
+    accessibility = accessibilityTags.map((tag) => tag.id);
   }
-}
+  let custom: ObjectId[] = [];
+  if (query.tags) {
+    const normalTags = await TagModel.find({
+      name: { $in: query.tags },
+      type: "custom",
+    });
+    if (query.tags.length !== normalTags.length) {
+      throw new GenericGameErrorException(
+        "One or more of the custom tags do not exist.",
+      );
+    }
+    custom = normalTags.map((tag) => tag.id);
+  }
```

Comment:
Consider refactoring conditional checks for fields in `query` for readability, I think I left a [comment](https://github.com/GTBitsOfGood/jennifer-anns-group/pull/39#discussion_r1501647268) about this last time, applies to all of these conditional checks